src/index.ts

```diff
@@ -19,12 +19,20 @@ interface Event<T> extends State {
 interface OnResize<T> {
   (event: Event<T>): void;
 }
+interface SetState {
+  (previous: State, next: State): State;
+}
 type Breakpoints = { [key: string]: number };
 export interface Options<T> {
   ref?: RefObject<T>;
   useBorderBoxSize?: boolean;
   breakpoints?: Breakpoints;
   onResize?: OnResize<T>;
+  /** If set, it will only update the state when a breakpoint is changed. */
+  onlyUpdateOnBreakpointChange?: boolean;
+  /** If you wish to conditionally update the internal state, for instance to
+   * reduce rerenders conditionally, you may use this custom-functionality */
+  customSetState?: SetState;
```

Comment:
@runar-rkmedia How about the following idea?

```js
const {/* ... */} = useDimensions({
  shouldUpdate: (prevState, nextState) => prevState.width > 300,
});
```

Reply:
Yes, I agree, the naming here is a lot better.

Reply:
@runar-rkmedia I also think if we can return a `boolean` to tell the state to update or not, like the `shouldComponentUpdate` lifecycle of React.